pkg/messaging/pair/client.go

```diff
@@ -0,0 +1,83 @@
+package pair
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+
+	"go.nanomsg.org/mangos/v3/protocol/pair"
+	"log"
+)
+
+func StartMessagingPairClient(ctx context.Context, nanomsgURL string, requestPair chan RequestPair, responsePair chan ResponsePair) error {
+	socket, err := pair.NewSocket()
+	if err != nil {
+		log.Printf("failed to get new pair socket: %v", err)
+		return err
+	}
```

Comment:
Do not forget to close socket if any error happens in `socket.Dial`. Or you can do smth like that
```suggestion
func StartMessagingPairClient(ctx context.Context, nanomsgURL string, requestPair chan RequestPair, responsePair chan ResponsePair) (err error) {
	socket, err := pair.NewSocket()
	if err != nil {
		log.Printf("failed to get new pair socket: %v", err)
		return err
	}
	defer func() {
		if err != nil {
			if err := socket.Close(); err != nil {
				log.Printf("failed to close socket: %v", err)
			}
		}
	}()
```